### PR TITLE
DEV-6172: Refactor create_scope_id into shell function

### PR DESCRIPTION
**Pull request by thomasdotred**
# Pull Request Checklist

- [ ] Read the [contributing guidelines](https://github.com/finbourne/lusid-python-tools/blob/master/docs/CONTRIBUTING.md)
- [ ] Tests pass

# Description of the PR

This PR encapsulates the time-based ID creation into another function, not specific to scope. Create_scope_id is left as a shell for compatibility with existing usages

### CODE REVIEW

File: lusidtools/cocoon/utilities.py [hunk 1]
```diff
@@ -19,6 +19,7 @@
 import time as default_time
 from lusidtools.cocoon.validator import Validator
 import types
+import uuid
```
Comment (slemasne) [inline lusidtools/cocoon/utilities.py:22]: Where is this used in the code? It looks like you're generating your own uid from the timestamp hex?

File: lusidtools/cocoon/utilities.py [hunk 2]
```diff
@@ -1354,8 +1355,26 @@ def create_scope_id(time_generator=None):
     # Multiply by 7 to get value to 100s of nano seconds
     timestamp = hex(int(test * 10000000.0))
```
Comment (slemasne) [inline lusidtools/cocoon/utilities.py:1356]: Perhaps change the `test` variable name to `current_time`

File: lusidtools/cocoon/utilities.py [hunk 3]
```diff
@@ -1354,8 +1355,26 @@ def create_scope_id(time_generator=None):
...
     # Create the scope id by joining the hex representation with dashes every 4 characters
-    scope_id = "-".join(timestamp[i : i + 4] for i in range(2, len(timestamp), 4))
-    return scope_id
+    uid = "-".join(timestamp[i : i + 4] for i in range(2, len(timestamp), 4))
```
Comment (slemasne) [inline lusidtools/cocoon/utilities.py:1358]: The `load_from_data_frame` function doesnt like `-` and it converts them to `-` but thats an issue with `load_from_data_frame` - we should probably fix that function to allow `-` as LUSID permits `-`

Comment (thomasdotred) [reply lusidtools/cocoon/utilities.py:1358]: Um whats the difference between the two `-` above? I cant see in the formatting!

File: lusidtools/cocoon/utilities.py (hunk 2 above)
Comment (thomasdotred) [reply lusidtools/cocoon/utilities.py:1356]: Changed

File: lusidtools/cocoon/utilities.py (hunk 1 above)
Comment (thomasdotred) [reply lusidtools/cocoon/utilities.py:22]: removed

File: lusidtools/cocoon/utilities.py (hunk 3 above)
Comment (slemasne) [reply lusidtools/cocoon/utilities.py:1358]: Underscore (_) versus hyphen (-).
The function like underscore.

Comment (thomasdotred) [reply lusidtools/cocoon/utilities.py:1357]: Ah ok, I must have miss-read as a hyphen

**Pull request merged**